app/src/main/java/ml/docilealligator/infinityforreddit/adapters/CommentsRecyclerViewAdapter.java

```diff
@@ -736,17 +716,9 @@ public void onFetchMoreCommentFailed() {
                                                 && parentComment.getFullName().equals(mVisibleComments.get(parentPosition).getFullName())) {
                                             if (mVisibleComments.get(parentPosition).isExpanded()) {
                                                 int commentPosition = mIsSingleCommentThreadMode ? holder.getBindingAdapterPosition() - 1 : holder.getBindingAdapterPosition();
-                                                int placeholderPosition = commentPosition;
-                                                if (commentPosition >= mVisibleComments.size() || commentPosition < 0 || !mVisibleComments.get(commentPosition).getFullName().equals(parentComment.getFullName())) {
-                                                    for (int i = parentPosition + 1; i < mVisibleComments.size(); i++) {
-                                                        if (mVisibleComments.get(i).getFullName().equals(parentComment.getFullName())) {
-                                                            placeholderPosition = i;
-                                                            break;
-                                                        }
-                                                    }
-                                                }
+                                                int placeholderPosition = findLoadMoreCommentsPlaceholderPosition(parentComment.getFullName(), commentPosition);
 
-                                                if (placeholderPosition >= mVisibleComments.size() || placeholderPosition < 0) {
```

Comment:
Is it a bug? `placeholderPosition >= mVisibleComments.size() || placeholderPosition < 0` does not make any sense haha.

Reply:
Yes, it should've been inverted. Fortunately that condition was always false because of the search above.
Now you'd think that it wouldn't update the button, but the text is updated unconditionally outside the check. This is a separate problem because holder can get bound to another placeholder by the time callback is triggered.

The only thing I don't understand here is why `mVisibleComments.get(placeholderPosition)` is not the same object as `mVisibleComments.get(parentPosition).getChildren().get(mVisibleComments.get(parentPosition).getChildren().size() - 1)`. Because scrolling up and down changes button's text back to "Loading". But then collapsing and expanding parent comment correctly displays "loading failed". So it means we have two Comment objects that represent the same placeholder that got out of sync.

However that's not all! If you repeat above steps again with the same comment, it will display "loading failed" after scrolling. So the two objects are now the same object! At least this is expected, because that's what `expandComments` should do. Why this isn't the case after initial load? No idea.